### PR TITLE
SPCTR-338-code-optimisation

### PR DESCRIPTION
**Pull request by rahulvarma722**
### Description

1. Remove extra properties parentProps from all blocks and keep them as props https://brainstormforce.atlassian.net/jira/software/projects/SPCTR/boards/1?selectedIssue=SPCTR-2399.
2. Remove setAttributes from all blocks and keep them in the HOC component https://brainstormforce.atlassian.net/browse/SPCTR-2790.
3. Remove unwanted code from form and form children blocks https://brainstormforce.atlassian.net/jira/software/projects/SPCTR/boards/1?selectedIssue=SPCTR-2399.
4. Remove substr from all blocks ( clientId.substr( 0, 8 ) ) to prevent extra process and keep block_id placed of this in 7 core blocks besides container block https://brainstormforce.atlassian.net/jira/software/projects/SPCTR/boards/1?assignee=62cf9daea94a6f9c0efeea37&selectedIssue=SPCTR-338.
5. Resolve reusable block issue in 7 core blocks besides container block https://brainstormforce.atlassian.net/jira/software/projects/SPCTR/boards/1?assignee=62cf9daea94a6f9c0efeea37&selectedIssue=SPCTR-2927.
 
### Checklist:
- [ ] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

**Review by adi3890 — APPROVED** [review]
Although code looks fine.
Testing need to be done thoroughly.

**Review by AbhijitNage123 — APPROVED** [review]

File: src/blocks/forms/child-blocks/phone/block.js [hunk 1]
```diff
@@ -8,10 +8,11 @@ import Edit from './edit';
 import save from './save';
 import { __ } from '@wordpress/i18n';
 import { registerBlockType } from '@wordpress/blocks';
-import deprecated from './deprecated';
 import PreviewImage from '@Controls/previewImage';
 import { applyFilters } from '@wordpress/hooks';
 import addCommonDataToSpectraBlocks from '@Controls/addCommonDataToSpectraBlocks';
+import Version from './deprecated/';
+import Version242 from './v2.4.2';
```
Comment (bsf-zanev) [inline src/blocks/forms/child-blocks/phone/block.js:15]: Let's rename these deprecated imports as well.

File: src/blocks/forms/child-blocks/radio/block.js [hunk 2]
```diff
@@ -7,11 +7,12 @@ import attributes from './attributes';
 import Edit from './edit';
 import save from './save';
 import { __ } from '@wordpress/i18n';
-import deprecated from './deprecated';
 import { registerBlockType } from '@wordpress/blocks';
 import PreviewImage from '@Controls/previewImage';
 import { applyFilters } from '@wordpress/hooks';
 import addCommonDataToSpectraBlocks from '@Controls/addCommonDataToSpectraBlocks';
+import Version from './deprecated/';
+import Version242 from './v2.4.2';
```
Comment (bsf-zanev) [inline src/blocks/forms/child-blocks/radio/block.js:15]: Let's rename these deprecated imports as well.

File: src/blocks/forms/child-blocks/select/block.js [hunk 3]
```diff
@@ -8,10 +8,11 @@ import Edit from './edit';
 import save from './save';
 import { __ } from '@wordpress/i18n';
 import { registerBlockType } from '@wordpress/blocks';
-import deprecated from './deprecated';
 import PreviewImage from '@Controls/previewImage';
 import { applyFilters } from '@wordpress/hooks';
 import addCommonDataToSpectraBlocks from '@Controls/addCommonDataToSpectraBlocks';
+import Version from './deprecated/';
+import Version242 from './v2.4.2';
```
Comment (bsf-zanev) [inline src/blocks/forms/child-blocks/select/block.js:15]: Let's rename these deprecated imports as well.

File: src/blocks/forms/child-blocks/textarea/block.js [hunk 4]
```diff
@@ -6,12 +6,13 @@ import UAGB_Block_Icons from '@Controls/block-icons';
...
 import Edit from './edit';
 import save from './save';
-import deprecated from './deprecated';
 import { __ } from '@wordpress/i18n';
 import { registerBlockType } from '@wordpress/blocks';
 import PreviewImage from '@Controls/previewImage';
 import { applyFilters } from '@wordpress/hooks';
 import addCommonDataToSpectraBlocks from '@Controls/addCommonDataToSpectraBlocks';
+import Version from './deprecated/';
+import Version242 from './v2.4.2';
```
Comment (bsf-zanev) [inline src/blocks/forms/child-blocks/textarea/block.js:15]: Let's rename these deprecated imports as well.

File: src/blocks/forms/child-blocks/toggle/block.js [hunk 5]
```diff
@@ -6,12 +6,14 @@ import UAGB_Block_Icons from '@Controls/block-icons';
...
 import save from './save';
-import deprecated from './deprecated';
 import { __ } from '@wordpress/i18n';
 import { registerBlockType } from '@wordpress/blocks';
 import PreviewImage from '@Controls/previewImage';
 import { applyFilters } from '@wordpress/hooks';
 import addCommonDataToSpectraBlocks from '@Controls/addCommonDataToSpectraBlocks';
+import Version from './deprecated/';
+import Version241 from './v2.4.1';
+import Version242 from './v2.4.2';
```
Comment (bsf-zanev) [inline src/blocks/forms/child-blocks/toggle/block.js:16]: Let's rename these deprecated imports as well.

File: src/blocks/forms/child-blocks/url/block.js [hunk 6]
```diff
@@ -6,12 +6,13 @@ import UAGB_Block_Icons from '@Controls/block-icons';
...
 import Edit from './edit';
 import save from './save';
-import deprecated from './deprecated';
 import { __ } from '@wordpress/i18n';
 import { registerBlockType } from '@wordpress/blocks';
 import PreviewImage from '@Controls/previewImage';
 import { applyFilters } from '@wordpress/hooks';
 import addCommonDataToSpectraBlocks from '@Controls/addCommonDataToSpectraBlocks';
+import Version from './deprecated/';
+import Version242 from './v2.4.2';
```
Comment (bsf-zanev) [inline src/blocks/forms/child-blocks/url/block.js:15]: Let's rename these deprecated imports as well.

**Review by bsf-zanev — COMMENTED** [review]
Few minor changes.

File: src/blocks/forms/child-blocks/accept/block.js [hunk 7]
```diff
@@ -14,6 +14,8 @@ import { applyFilters } from '@wordpress/hooks';
 import addCommonDataToSpectraBlocks from '@Controls/addCommonDataToSpectraBlocks';
 let acceptCommonData = {};
 acceptCommonData = applyFilters( 'uagb/forms-accept', addCommonDataToSpectraBlocks( acceptCommonData ) );
+import Version242 from './v2.4.2';
```
[the remark this answers is not in the document]
Comment (rahulvarma722) [reply src/blocks/forms/child-blocks/accept/block.js:17]: The deprecated file name changed.

File: src/blocks/forms/child-blocks/checkbox/block.js [hunk 8]
```diff
@@ -6,15 +6,16 @@ import UAGB_Block_Icons from '@Controls/block-icons';
...
-
 import { registerBlockType } from '@wordpress/blocks';
 import { __ } from '@wordpress/i18n';
 import PreviewImage from '@Controls/previewImage';
 import { applyFilters } from '@wordpress/hooks';
 import addCommonDataToSpectraBlocks from '@Controls/addCommonDataToSpectraBlocks';
 let checkboxCommonData = {};
 checkboxCommonData = applyFilters( 'uagb/forms-checkbox', addCommonDataToSpectraBlocks( checkboxCommonData ) );
+import Version from './deprecated/';
+import Version242 from './v2.4.2';
```
[the remark this answers is not in the document]
Comment (rahulvarma722) [reply src/blocks/forms/child-blocks/checkbox/block.js:17]: The deprecated file name changed.

File: src/blocks/forms/child-blocks/date/block.js [hunk 9]
```diff
@@ -15,6 +14,9 @@ import { applyFilters } from '@wordpress/hooks';
 import addCommonDataToSpectraBlocks from '@Controls/addCommonDataToSpectraBlocks';
 let dateCommonData = {};
 dateCommonData = applyFilters( 'uagb/forms-date', addCommonDataToSpectraBlocks( dateCommonData ) );
+import Version241 from './v2.4.1';
+import Version242 from './v2.4.2';
```
[the remark this answers is not in the document]
Comment (rahulvarma722) [reply src/blocks/forms/child-blocks/date/block.js:18]: The deprecated file name changed.

File: src/blocks/forms/child-blocks/email/block.js [hunk 10]
```diff
@@ -6,14 +6,16 @@ import UAGB_Block_Icons from '@Controls/block-icons';
...
-import deprecated from './deprecated';
 import { __ } from '@wordpress/i18n';
 import { registerBlockType } from '@wordpress/blocks';
 import PreviewImage from '@Controls/previewImage';
 import { applyFilters } from '@wordpress/hooks';
 import addCommonDataToSpectraBlocks from '@Controls/addCommonDataToSpectraBlocks';
 let emailCommonData = {};
 emailCommonData = applyFilters( 'uagb/forms-email', addCommonDataToSpectraBlocks( emailCommonData ) );
+import Version from './deprecated/';
+import Version242 from './v2.4.2/';
```
[the remark this answers is not in the document]
Comment (rahulvarma722) [reply src/blocks/forms/child-blocks/email/block.js:17]: The deprecated file name changed.

File: src/blocks/forms/child-blocks/name/block.js [hunk 11]
```diff
@@ -6,12 +6,13 @@ import UAGB_Block_Icons from '@Controls/block-icons';
...
 import Edit from './edit';
 import save from './save';
-import deprecated from './deprecated';
 import { __ } from '@wordpress/i18n';
 import { registerBlockType } from '@wordpress/blocks';
 import PreviewImage from '@Controls/previewImage';
 import { applyFilters } from '@wordpress/hooks';
 import addCommonDataToSpectraBlocks from '@Controls/addCommonDataToSpectraBlocks';
+import Version from './deprecated/';
+import Version242 from './v2.4.2/';
```
[the remark this answers is not in the document]
Comment (rahulvarma722) [reply src/blocks/forms/child-blocks/name/block.js:15]: The deprecated file name changed.

File: src/blocks/forms/child-blocks/phone/block.js (hunk 1 above)
Comment (rahulvarma722) [reply src/blocks/forms/child-blocks/phone/block.js:15]: The deprecated file name changed.

File: src/blocks/forms/child-blocks/radio/block.js (hunk 2 above)
Comment (rahulvarma722) [reply src/blocks/forms/child-blocks/radio/block.js:15]: The deprecated file name changed.

File: src/blocks/forms/child-blocks/select/block.js (hunk 3 above)
Comment (rahulvarma722) [reply src/blocks/forms/child-blocks/select/block.js:15]: The deprecated file name changed.

File: src/blocks/forms/child-blocks/textarea/block.js (hunk 4 above)
Comment (rahulvarma722) [reply src/blocks/forms/child-blocks/textarea/block.js:15]: The deprecated file name changed.

File: src/blocks/forms/child-blocks/toggle/block.js (hunk 5 above)
Comment (rahulvarma722) [reply src/blocks/forms/child-blocks/toggle/block.js:16]: The deprecated file name changed.

File: src/blocks/forms/child-blocks/url/block.js (hunk 6 above)
Comment (rahulvarma722) [reply src/blocks/forms/child-blocks/url/block.js:15]: The deprecated file name changed.

File: blocks-config/uagb-controls/addInitialAttr.js [hunk 12]
```diff
@@ -0,0 +1,189 @@
...
+				'uagb/restaurant-menu',
+				'uagb/social-share',
+                'uagb/content-timeline',
+			];
+
+			const listOfIsHtml = [ 'uagb/cf7-styler', 'uagb/gf-styler' ];
+
+			const listOfEditorInnerblocksPreview = [ 'uagb/countdown' ];
+
+			const listOfAllTaxonomyStore = [ 'uagb/post-carousel', 'uagb/post-grid', 'uagb/post-masonry' ];
```
[the remark this answers is not in the document]
Comment (rahulvarma722) [reply blocks-config/uagb-controls/addInitialAttr.js:63]: These variables are just for temporary purposes once all things will be centralise we will remove all variables from useEffect.

File: src/blocks/post/post-carousel/edit.js [hunk 13]
```diff
@@ -425,27 +425,25 @@ const UAGBPostCarousel = ( props ) => {
...
 			setAttributes( { dotsMarginTopMobile: columnGapMobile } );
 		}
-
-		setAttributes( { allTaxonomyStore: undefined } );
 	}, [] );
 
 	useEffect( () => {
 		if ( equalHeight ) {
-			uagb_carousel_height( props.clientId.substr( 0, 8 ) ); // eslint-disable-line no-undef
+			uagb_carousel_height( clientId.substr( 0, 8 ) ); // eslint-disable-line no-undef
```
Comment (aapatre) [inline src/blocks/post/post-carousel/edit.js:432]: Declare in `eslintrc.js` so that no eslint disable is required

**Review by aapatre — APPROVED** [review]
Pre-approved with one minor comment.

File: src/blocks/post/post-carousel/edit.js (hunk 13 above)
Comment (rahulvarma722) [reply src/blocks/post/post-carousel/edit.js]: Declared in `eslintrc.js` file

**Pull request merged**